File: app/src/main/java/fit/tdc/edu/vn/cafemanagement/data/model/Payment.java
package fit.tdc.edu.vn.cafemanagement.data.model;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Payment {
    private String id;
    private String storeID;
    private String tableID;
    private String confirmedEmployeeID;
    private LocalDateTime paidTime;
    private String price;
    private String stateID;
    private String formatID;
}

File: app/src/main/java/fit/tdc/edu/vn/cafemanagement/data/model/kotlin/Payment.kt
package fit.tdc.edu.vn.cafemanagement.data.model.kotlin

import com.google.firebase.Timestamp
import com.google.firebase.firestore.Exclude
import fit.tdc.edu.vn.cafemanagement.data.model.FirestoreModel
import fit.tdc.edu.vn.cafemanagement.data.model.order.Order

data class Payment(
    var storeId: String? = null,
    var storeName: String? = null,
    var tableId: String? = null,
    var confirmedEmployeeId: String? = null,
    var confirmedEmployeeName: String? = null,
    var paidTime: Timestamp? = null,
    var state: State? = null,
    @get:Exclude var orderList: List<Order> = listOf(),
    var total: Long = 0
) : FirestoreModel() {
    enum class State {
        ORDERING,
        ORDERED,
        PAID
    }
}
